tests/rest/client/test_retention.py

```diff
@@ -295,6 +299,23 @@ def test_state_policy(self) -> None:
 
         self._test_retention(room_id, expected_code_for_first_event=404)
 
+    @unittest.override_config({"retention": {"enabled": False}})
+    def test_visibility_when_disabled(self) -> None:
```

Comment:
I think describing what this test is doing is helpful:

```suggestion
    def test_visibility_when_disabled(self) -> None:
        """Retention policies should be ignored when the retention feature is disabled."""
```